### ISSUE TITLE
Only push Git commit in trigger.py

### ISSUE THREAD

**Issue by gctucker** [kernelci/kernelci-pipeline#3]
The current `trigger.py` script provides a proof-of-concept implementation for submitting new revisions to the API.  A few things need to be modified in order to be able to use it in a real-world pipeline:

* Rather than updating a local full Git checkout, use `kernelci.build.get_branch_head()` to get the latest git commit for the build config being monitored (e.g. `mainline`)
* Send a GET query to the API to see if there is any Node object with a revision matching the git commit found.  If there is one, then stop at this point and try again later.
* If there isn't a Node with a revision with the git commit found, submit one with an empty `describe` and empty `status`.  The API will then generate a pub/sub event.

This should be all that the `trigger.py` script should do.  Another client will then be listening for events when a new revision has been added to push a source tarball and update the revision information with the git describe (requires a local checkout) and set the status to `True` if it worked.  Then other clients will be running tests that rely on the source tarball etc.

Depends on https://github.com/kernelci/kernelci-api/issues/66 to only create new Node using `POST`.

**Comment by JenySadadia** [comment]
Hello @gctucker, 
For the below task: 
```
Submit a Node object without describe in the revision to the API and without status value, which will return an error if the revision already exists and ignore it or create a new one and generate a pub/sub event if it's a new one
```
I tried to POST node without 'describe'. kernelci-api is not able to process the request because of the missing property.
```
trigger     | Sending revision node to API: df0cc57e057f18e44dac8e6c18aba47ab53202f9
trigger     | node {'name': 'checkout', 'status': True, 'revision': {'tree': 'mainline', 'url': 'https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git', 'branch': 'master', 'commit': 'df0cc57e057f18e44dac8e6c18aba47ab53202f9'}}
trigger     | 422 Client Error: Unprocessable Entity for url: http://172.17.0.1:8001/node
trigger     | Traceback (most recent call last):
trigger     |   File "/home/kernelci/./pipeline/trigger.py", line 101, in <module>
trigger     |     status = opts.command(configs, opts)
trigger     |   File "/home/kernelci/./pipeline/trigger.py", line 89, in __call__
trigger     |     _run_trigger(args, build_config, db)
trigger     |   File "/home/kernelci/./pipeline/trigger.py", line 56, in _run_trigger
trigger     |     resp_obj = db.submit({'node': node})[0]
trigger     |   File "/home/kernelci/.local/lib/python3.10/site-packages/kernelci/data/kernelci_api.py", line 79, in submit
trigger     |     raise(ex)
trigger     |   File "/home/kernelci/.local/lib/python3.10/site-packages/kernelci/data/kernelci_api.py", line 76, in submit
trigger     |     resp = self._post(path, json.dumps(item))
trigger     |   File "/home/kernelci/.local/lib/python3.10/site-packages/kernelci/data/kernelci_api.py", line 50, in _post
trigger     |     resp.raise_for_status()
trigger     |   File "/home/kernelci/.local/lib/python3.10/site-packages/requests/models.py", line 960, in raise_for_status
trigger     |     raise HTTPError(http_error_msg, response=self)
trigger     | requests.exceptions.HTTPError: 422 Client Error: Unprocessable Entity for url: http://172.17.0.1:8001/nodeINFO[2022-01-10T09:44:29.067773600+05:30] ignoring event                                container=ae52f455b08134e8c755d615edb6fc786215d54c1cf14c9389b601db5c4825cc module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
INFO[2022-01-10T09:44:29.067961800+05:30] shim disconnected                             id=ae52f455b08134e8c755d615edb6fc786215d54c1cf14c9389b601db5c4825cc
trigger exited with code 1
```

**Comment by JenySadadia** [comment]
For the below task,
```
Rather than updating a local full Git checkout, use kernelci.build.get_branch_head() to get the latest git commit for the build config being monitored (e.g. mainline)
```
We need to fetch the latest commit SHA using ```get_branch_head()```. We need to pass the commit SHA to ```update_repo()``` to checkout the commit after cloning repo. We need to modify build.py for it. 
Is this the correct approach ? @gctucker

**Comment by gctucker** [comment]
Or maybe the git commit number could be sent, and if the client making the tarball finds a newer git commit then it will update the node with or create a new one.  We just need to make sure we don't have conflicts between clients running in parallel.

**Comment by JenySadadia** [comment]
> I think there's still a problem with this approach, as the git commit might disappear between when the revision is discovered and when the tarball is created. So we should follow the same approach as what is done with the current Jenkins pipeline. It seems like we need some document to outline all these steps.
> 
> The `trigger.py` client could send a Node object with just the tree and branch name but without a commit number to signal there is a new revision available, then the other client making the tarball would update that revision with the git commit.

@gctucker Could you please describe Jenkins pipeline in detail or provide me any reference documentation for the same ?

**Comment by JenySadadia** [comment]
I guess we need to add logic to allow to create Node object without the 'describe' property.
Please correct me if I have misunderstood. @gctucker

**Comment by gctucker** [comment]
@JenySadadia I've updated the description of this task, I think we can basically stick to something similar that was initially planned.  It should be completely fine this way for the initial deployment on staging.  We just have a few known limitations which we'll need to address before this can be used in production, and these should be covered by follow-up issues.  In particular:
* Some branches get force-pushed frequently, so the git commit might disappear between the moment when a new revision is pushed to the API and when another client tries to use it (typically true with stable-rc branches and some maintainers).  As long as we use mainline and linux-next on staging, we won't have this problem.  To solve this issue, we may use a different pub/sub channel and not push a Node in the trigger client (to be better defined in a separate issue).
* If several trigger clients are running at the same time, there will be a race condition between querying the API for revisions that have already been submitted and submitting a new one.  It shouldn't be a problem with the initial staging deployment, but it will need to be fixed before production deployment.  This issue would then move from the trigger client to the one creating tarballs with the solution explained above, so it will need to be addressed in any case.  It seems that we can rely on Mongo DB indexes to enforce some field combinations to be unique (revision), then the API would return an error when trying to push a revision that already exists (also to be defined in a separate issue).
